src/bokeh/models/comparisons.py

```diff
@@ -0,0 +1,123 @@
+#-----------------------------------------------------------------------------
+# Copyright (c) Anaconda, Inc., and Bokeh Contributors.
+# All rights reserved.
+#
+# The full license is in the file LICENSE.txt, distributed with this software.
+#-----------------------------------------------------------------------------
+''' Represent comparisons to be computed on the client (browser) side by
+BokehJS. These comparisons may be useful for specifying how DataTable columns
+are sorted.
+
+'''
+
+#-----------------------------------------------------------------------------
+# Boilerplate
+#-----------------------------------------------------------------------------
+from __future__ import annotations
+
+import logging # isort:skip
+log = logging.getLogger(__name__)
+
+#-----------------------------------------------------------------------------
+# Imports
+#-----------------------------------------------------------------------------
+
+# Bokeh imports
+from ..core.has_props import abstract
+from ..core.properties import (
+    AnyRef,
+    Bool,
+    Dict,
+    String,
+)
+from ..model import Model
+
+#-----------------------------------------------------------------------------
+# Globals and constants
+#-----------------------------------------------------------------------------
+
+__all__ = (
+    'Comparison',
+    'CustomJSCompare',
+    'NanCompare',
+)
+
+#-----------------------------------------------------------------------------
+# General API
+#-----------------------------------------------------------------------------
+
+@abstract
+class Comparison(Model):
+    ''' Base class for ``Comparison`` models that represent a comparison
+    to be carried out on the client-side.
+
+    JavaScript implementations should implement the following method:
+
+    .. code-block:: typescript
```

Comment:
As far as I know, there is no highligting for `typescript`. But the code block looks good to me and I would keep this as it is.

![grafik](https://github.com/user-attachments/assets/25ebc27e-29d4-449a-885a-7cf1edc80175)